docs/installation.md

```diff
@@ -0,0 +1,111 @@
+
+# Installation
+
+Arlon CLI downloads are provided on GitHub. The CLI is not a self-contained standalone executable though. 
+It is required to point the CLI to a management cluster and set up the Arlon controller in this management cluster.
+
+We plan to add a command line script to simplify initial setup. 
+For a quickstart minimal demonstration setup, follow the instructions to set up a KIND based testbed [here](https://github.com/arlonproj/arlon/blob/main/testing/README.md)
+
+Please follow the manual instructions in this section for a customised setup.
+
+## Management cluster
+
+You can use any Kubernetes cluster that you have admin access to. Ensure:
+- `kubectl` is in your path
+- `KUBECONFIG` is pointing to the right file and the context set properly
+
+## ArgoCD
+
+- Follow steps 1-4 of the [ArgoCD installation guide](https://argo-cd.readthedocs.io/en/stable/getting_started/) to install ArgoCD onto your management cluster.
+After this step, you should be logged in as `admin` and a config file was created at `${HOME}/.config/argocd/config`
+- Create your workspace repository in your git provider if necessary, then register it.
+  Example: `argocd repo add https://github.com/myname/arlon_workspace --username myname --password secret`.
+   -  Note: type `argocd repo add --help` to see all available options.
+   -  For Arlon developers, this is not your fork of the Arlon source code repository, 
+       but a separate git repo where some artifacts like profiles created by Arlon will be stored. 
+- Highly recommended: [configure a webhook](https://argo-cd.readthedocs.io/en/stable/operator-manual/webhook/)
+  to immediately notify ArgoCD of changes to the repo. This will be especially useful
+  during the tutorial. Without a webhook, repo changes may take up to 3 minutes
+  to be detected, delaying cluster configuration updates. 
+- [Create a local user](https://argo-cd.readthedocs.io/en/stable/operator-manual/user-management/) named `arlon` with the `apiKey` capability.
+  This involves editing the `argocd-cm` ConfigMap using `kubectl`.
+- Adjust the RBAC settings to grant admin permissions to the `arlon` user.
+  This involves editing the `argocd-rbac-cm` ConfigMap to add the entry
+  `g, arlon, role:admin` under the `policy.csv` section. Example:
+```
+apiVersion: v1
+data:
+  policy.csv: |
+    g, arlon, role:admin
+kind: ConfigMap
+[...]
+```
+- Generate an account token: `argocd account generate-token --account arlon`
+- Make a temporary copy of the config file: `cp ${HOME}/.config/argocd/config /tmp/config` then
```

Comment:
Shouldn't this point be changed to make a temp config file from testing/argocd-config-for-controller.template.yaml?

Reply:
Created a separate issue to track this. https://github.com/arlonproj/arlon/issues/310